transformer/test_task.py

```diff
@@ -2,256 +2,140 @@
-        assert action.startswith(" \n def function():\n  if True:\n   print(True)")
-
-
-class TestReplaceURL:
-    def test_it_creates_a_locust_request_when_there_is_none(self):
-        task = Task(name="some name", request=MagicMock())
-
-        modified_task = Task.replace_url(task, "")
-
-        assert modified_task.locust_request
-
-    def test_it_returns_a_task_with_the_replaced_url(self):
-        locust_request = LocustRequest(
-            method=MagicMock(), url=MagicMock(), headers=MagicMock()
-        )
-        task = Task(
-            name="some name", request=MagicMock(), locust_request=locust_request
-        )
-        expected_url = 'f"http://a.b.c/{some.value}/"'
-
-        modified_task = Task.replace_url(task, expected_url)
-
-        assert modified_task.locust_request.url == expected_url
+class TestTask:
+    class TestFromRequests:
+        def test_it_returns_a_task(self):
+            request = MagicMock()
+            request.timestamp = 1
+            second_request = MagicMock()
+            second_request.timestamp = 2
+            assert all(
+                isinstance(t, Task)
+                for t in Task.from_requests([request, second_request])
+            )
+
+        @patch("builtins.open")
+        def test_it_doesnt_create_a_task_if_the_url_is_on_the_blacklist(
+            self, mock_open
+        ):
+            mock_open.return_value = io.StringIO("amazon")
+            request = MagicMock()
+            request.url = MagicMock()
+            request.url.netloc = "www.amazon.com"
+            task = Task.from_requests([request])
+            assert len(list(task)) == 0
+
+        @patch("builtins.open")
+        def test_it_creates_a_task_if_the_path_not_host_is_on_the_blacklist(
+            self, mock_open
+        ):
+            mock_open.return_value = io.StringIO("search\namazon")
+            request = MagicMock()
+            request.url = urlparse("https://www.google.com/search?&q=amazon")
+            task = Task.from_requests([request])
+            assert len(list(task)) == 1
+
+    class TestReplaceURL:
+        def test_it_creates_a_locust_request_when_there_is_none(self):
+            task = Task(name="some name", request=MagicMock())
+
+            modified_task = Task.replace_url(task, "")
+
+            assert modified_task.locust_request
+
+        def test_it_returns_a_task_with_the_replaced_url(self):
+            locust_request = LocustRequest(
+                method=MagicMock(), url=MagicMock(), headers=MagicMock()
+            )
+            task = Task(
+                name="some name", request=MagicMock(), locust_request=locust_request
+            )
+            expected_url = 'f"http://a.b.c/{some.value}/"'
+
+            modified_task = Task.replace_url(task, expected_url)
+
+            assert modified_task.locust_request.url == expected_url
+
+
+class TestTask2:
+    class TestFromTask:
+        def test_without_locust_request(self):
```

Comment:
I'm in favour of keeping the tests as granular as possible, both `test_without_locust_request` and  `test_with_locust_request` seem to be too general. Would it make sense to split these tests based on what they check, rather than - in this particular example - checking everything related to `locust_request` in a single test?